tests/ops/qubit/test_matrix_ops.py

```diff
@@ -540,12 +571,47 @@ def test_broadcasted_two_qubit_qubit_unitary_decomposition_raises_error(self):
         with pytest.raises(DecompositionUndefinedError, match="QubitUnitary does not support"):
             qml.QubitUnitary(U, wires=[0, 1]).decomposition()
 
-    def test_qubit_unitary_decomposition_multiqubit_invalid(self):
-        """Test that QubitUnitary is not decomposed for more than two qubits."""
-        U = qml.Toffoli(wires=[0, 1, 2]).matrix()
+    @pytest.mark.parametrize(
+        "U, wires",
+        [
+            (qml.matrix(qml.CRX(2, wires=[1, 0])), [0, 1]),
+            (qml.matrix(qml.QFT(wires=[0, 1, 2, 3, 4])), [0, 1, 2, 3, 4]),
+            (qml.matrix(qml.RX(1, 0) @ qml.CRY(2, [1, 2])), [0, 1, 2]),
+            (qml.matrix(qml.GroverOperator([0, 1, 2, 3, 4, 5])), [0, 1, 2, 3, 4, 5]),
+        ],
+    )
+    def test_correctness_decomposition(self, U, wires):
+        """Tests that the decomposition is correct"""
+
+        ops_decompostion = qml.QubitUnitary.compute_decomposition(U, wires=wires)
+
+        assert qml.math.allclose(
+            U, qml.matrix(qml.prod(*ops_decompostion[::-1]), wire_order=wires), atol=1e-7
+        )
+
+    def test_compute_udv(self):
+        """Test the helper function `_compute_udv` used in the QubitUnitary decomposition."""
+
+        a = qml.matrix(qml.RY(1, 0))
+        b = qml.matrix(qml.RX(2, 0))
```

Comment:
If it's easy, I would suggest adding one test here with matrices on multiple qubits.

Reply:
Added 2 and 3 qubits examples 👍